### PR TITLE
Ensure CWPSearchBoostExtension doesn't expect MetaData fields to be present

### PR DESCRIPTION
**Pull request by scott1702**
Fixes https://github.com/silverstripe/cwp-search/issues/8

### CODE REVIEW

File: src/Extensions/CwpSearchBoostExtension.php [hunk 1]
```diff
@@ -31,12 +32,7 @@ class CwpSearchBoostExtension extends DataExtension
      */
     public function updateCMSFields(FieldList $fields)
     {
-        parent::updateCMSFields($fields);
```
Comment (scott1702) [inline src/Extensions/CwpSearchBoostExtension.php:34]: Didn't think this was necessary but let me know if I'm wrong :)

File: src/Extensions/CwpSearchBoostExtension.php [hunk 2]
```diff
@@ -31,14 +32,7 @@ class CwpSearchBoostExtension extends DataExtension
...
-
-        // Rename metafield
-        $meta = $fields->fieldByName('Root.Main.Metadata');
-        if ($meta) {
-            $meta->setTitle(_t(__CLASS__ . '.PAGEINFO', 'Page info and SEO'));
-        }
-
+        $pageInfoTitle = _t(__CLASS__ . '.PAGEINFO', 'Page info and SEO');
```
Comment (robbieaverill) [inline src/Extensions/CwpSearchBoostExtension.php:35]: Maybe you could put this down with the rest of the code

File: src/Extensions/CwpSearchBoostExtension.php [hunk 3]
```diff
@@ -52,6 +46,23 @@ public function updateCMSFields(FieldList $fields)
         $boostField = TextareaField::create('SearchBoost', $boostTitle)
             ->setRightTitle($boostNote)
             ->setDescription($boostDescription);
-        $fields->insertBefore('MetaDescription', $boostField);
+
+        if ($meta = $fields->fieldByName('Root.Main.Metadata')) {
+            // Rename metafield if it exists
+            $meta->setTitle($pageInfoTitle);
+            $fields->insertBefore('MetaDescription', $boostField);
```
Comment (robbieaverill) [inline src/Extensions/CwpSearchBoostExtension.php:53]: This line would previously be run regardless of whether the Metadata field exists, but it's now only run when it does exist. Perhaps you could move this back up to be underneath the `$boostField` definition

**Review by robbieaverill — COMMENTED** [review]
I've left a minor comment. Would you mind giving a quick summary of how this fixes the problem? To be honest I'm not really sure what the bug is in the first place. As far as I can tell, the only thing that line 38/39 is doing is changing the title of a form field

**Comment by robbieaverill** [comment]
I've run through this in person with @scott1702.

When the Metadata composite field doesn't exist, the search boost field doesn't get added at all. This PR means that when the Metadata composite doesn't exist, it creates one and adds the boost field to it so it shows up in the page CMS view.

LGTM after all!

**Pull request merged**